packages/astro/src/vite-plugin-config-alias/index.ts

```diff
@@ -72,16 +72,19 @@ export default function configAliasVitePlugin({
 				};
 			}
 		},
-		resolveId(id) {
+		async resolveId(id, importer, options) {
 			if (id.startsWith('.') || id.startsWith('/')) return;
 
 			// Handle baseUrl mapping for non-relative and non-root imports.
 			// Since TypeScript only applies `baseUrl` autocompletions for files that exist
 			// in the filesystem only, we can use this heuristic to skip resolve if needed.
 			const resolved = path.posix.join(resolvedBaseUrl, id);
-			if (fs.existsSync(resolved)) {
```

Comment:
When importing TypeScript files without the `.ts` extension, this condition is always false, as `/path/to/ts-flie` doesn't exist.
I thought about checking for TypeScript files but, I realized that there could be other file types that could be resolved when not specifying the extension, maybe via vite plugins pushed before the ts resolver?

Reply:
Great call!